src/main/java/de/tum/cit/aet/artemis/core/util/HttpRequestUtils.java

```diff
@@ -41,4 +51,131 @@ public static Optional<IPAddress> getIpAddressFromRequest(@NotNull HttpServletRe
+            return Browser.APPLE_SAFARI;
+        }
+        else if (secureClientHintsUserAgent.contains("Brave")) {
+            return Browser.BRAVE;
+        }
+        else if (userAgent.contains("Vivaldi")) {
+            return Browser.VIVALDI;
+        }
+        else if (userAgent.contains("DuckDuckGo")) {
+            return Browser.DUCKDUCKGO;
+        }
+
+        log.warn("Could not detect browser name from user agent: {}, secure user agent: {}", userAgent, secureClientHintsUserAgent);
+        return null;
+    }
+
+    /**
+     * Determines the operating system based on the {@code User-Agent} and {@code Sec-Ch-Ua-Platform} headers in the given {@link HttpServletRequest}.
+     * <p>
+     * Logs a warning if the operating system cannot be detected.
+     *
+     * @param request the HTTP request
+     * @return the detected {@link OperatingSystem}, or {@code null} if the operating system cannot be determined
+     */
+    private static OperatingSystem getOperatingSystem(@NotNull HttpServletRequest request) {
+        String userAgent = getHeaderValue(request, HttpHeaders.USER_AGENT);
+        String secureClientHintsUserAgentPlatform = getHeaderValue(request, "Sec-Ch-Ua-Platform");
+
+        if (userAgent.contains("Windows")) {
+            return OperatingSystem.WINDOWS;
+        }
+        else if (userAgent.contains("iPhone") || userAgent.contains("iPad") || userAgent.contains("iOS")) {
+            return OperatingSystem.IOS;
+        }
+        else if (userAgent.contains("Macintosh") || userAgent.contains("Mac OS X")) {
+            return OperatingSystem.MACOS;
+        }
+        else if (userAgent.contains("Android")) {
+            return OperatingSystem.ANDROID;
+        }
+        else if (userAgent.contains("Linux")) {
+            return OperatingSystem.LINUX;
+        }
+
+        log.warn("Could not detect operating system from user agent: {}, secure user agent platform: {}", userAgent, secureClientHintsUserAgentPlatform);
+        return null;
+    }
+
+    /**
+     * Extracts the client environment from the given {@link HttpServletRequest}.
+     * <p>
+     * The method determines the browser, operating system, or application type (iOS or Android) based on the request headers.
+     *
+     * @param request the HTTP request
+     * @return a {@link ClientEnvironment} object containing the detected client environment, or {@code null} if no environment can be determined
+     */
+    public static ClientEnvironment getClientEnvironment(@NotNull HttpServletRequest request) {
+        Browser browserName = getBrowserName(request);
+        OperatingSystem operatingSystem = getOperatingSystem(request);
+        if (browserName != null) {
+            return new ClientEnvironment(browserName, operatingSystem, null);
+        }
+
+        String userAgent = request.getHeader(HttpHeaders.USER_AGENT);
+
+        if (userAgent == null || userAgent.isEmpty()) {
```

Comment:
```suggestion
        if (userAgent.isEmpty()) {
```
 if you use the new helper like suggested above you can simplify here too :)